assets/src/blocks/Gallery/useGalleryImages.js

```diff
@@ -0,0 +1,58 @@
+import {useState, useEffect} from '@wordpress/element';
+import {fetchJson} from '../../functions/fetchJson';
+import {addQueryArgs} from '../../functions/addQueryArgs';
+import {getAbortController} from '../../functions/getAbortController';
+
+const {apiFetch} = wp;
+
+const GALLERY_IMAGE_SIZES = {
+  slider: 'retina-large',
+  'three-columns': 'medium_large',
+  grid: 'large',
+};
+
+export const useGalleryImages = ({multiple_image, gallery_block_focus_points}, layout, baseUrl = null) => {
+  const [images, setImages] = useState([]);
+  const [controller, setController] = useState();
+
+  const imageSize = GALLERY_IMAGE_SIZES[layout];
+
+  const loadPage = async () => {
+    const args = {
+      image_size: imageSize,
+      multiple_image,
+      gallery_block_focus_points,
+    };
+
+    try {
+      // eslint-disable-next-line no-shadow
+      const images = baseUrl ?
+        await fetchJson(`${baseUrl}/wp-json/${addQueryArgs('planet4/v1/get-gallery-images', args)}`) :
```

Comment:
I get this error below in the console, shows you are not registering the route in the master-theme.

<img width="740" alt="Screenshot 2023-12-20 at 10 47 15" src="https://github.com/greenpeace/planet4-master-theme/assets/38656549/fc52e39e-0476-47c0-9ebb-83feee72f9d1">